File: BookStore/Models/OrderDetail.cs
namespace BookStore.Models
{
    public class OrderDetail
    {
        public int OrderDetailID { get; set; } // PK 
        public int Quantity { get; set; }   

        //FK
        public int OrderID { get; set; }    
        public int BookID { get; set; }

        //Navigation properties

        public Order Order { get; set; } = null!; // many - to - one
        public Book Book { get; set; } = null!; // many - to - one
       
    }
}

File: client/src/Models/OrderDetail.ts
export interface OrderDetail {
  orderDetailID: number;
  quantity: number;
  orderID: number;
  bookID: number;
  bookName: string;
  bookPrice: number;
}

// public class OrderDetailDTO
// {
//     public int OrderDetailID { get; set; } // PK
//     public int Quantity { get; set; }

//     //FK
//     public int OrderID { get; set; }
//     public int BookID { get; set; }

//     public string BookName { get; set; } = string.Empty;

//     public decimal BookPrice { get; set; }
// }
